telicent-graph-schema/src/main/resources/io/telicent/jena/graphql/schemas/telicent/graph/ies.graphqls

```diff
@@ -41,13 +41,39 @@ type NonDirectionalRel {
     entity: Node!
 }
 
+enum SearchType {
+    QUERY,
+    TERM,
+    PHRASE,
+    WILDCARD
+}
+
+type SearchResults {
+    searchTerm: String!
+    searchType: SearchType!
+    limit: Int!
+    offset: Int!
+    maybeMore: Boolean!
+    nodes: [Node]
+}
+
 type Query {
     search(
         graph: String
         searchTerm: String!
+        searchType: SearchType
+        limit: Int
+        offset: Int
+        typeFilter: String
+    ): [Node] @deprecated(reason: "Use `searchV2` which offers richer response schema")
```

Comment:
@davec504 Since the existing `search` query only returned an array of nodes I couldn't change the response schema to include an indicator of whether there's further pages of results available as it would be a breaking change.  So following [GraphQL best practise](https://graphql.org/learn/best-practices/#versioning) I've instead introduced a new `searchV2` query with richer `SearchResults` response schema that can be adopted as applications are ready to do so.  Please take a look at the `SearchResults` type and let me know if that's going to fulfil the FE application needs.

Aside - I also added support for additional arguments - `searchType` and `typeFilter` - into both the old and new queries since they are optional arguments so not a breaking change so that we can expose those capabilities of search via GraphQL

Reply:
We could also name this something else like `searchWithMetadata` if that's preferable to `searchV2`?

Reply:
Looks good to me. I'm happy with either naming convention I don't really mind.

Reply:
@rvesse now though it is possible to filter the results by type, is it possible to get a list of what types the users can filter by? Would this have any negative performance impacts?

Reply:
Yes, though probably wants a separate ticket and PR

It would require a separate request to the Search API so there's some performance implication there